File: reactor/java-reactor/src/main/java/com/example/reactor/javareactor/dto/response/UserResponse.java
package com.example.reactor.javareactor.dto.response;

import com.example.reactor.javareactor.entity.User;
import lombok.Builder;

@Builder
public record UserResponse(
    String id,
    String email,
    String name
) {
    public static UserResponse from(User user) {
        return UserResponse.builder()
                           .id(user.getId())
                           .email(user.getEmail())
                           .name(user.getName())
                           .build();
    }
}

File: reactor/kotlin-reactor/src/main/kotlin/com/example/reactor/kotlinreactor/dto/response/UserResponse.kt
package com.example.reactor.kotlinreactor.dto.response

import com.example.reactor.kotlinreactor.entity.User

data class UserResponse(
    val id: String,
    val email: String,
    val name: String
) {
    companion object {
        operator fun invoke(user: User): UserResponse =
            with(user) {
                UserResponse(
                    id = id!!,
                    email = email,
                    name = name
                )
            }
    }
}
